packages/wifi_scan/lib/wifi_scan.dart

```diff
@@ -18,73 +18,66 @@ class WiFiScan {
   final _channel = const MethodChannel('wifi_scan');
   final _scannedResultsAvailableChannel =
       const EventChannel('wifi_scan/onScannedResultsAvailable');
-  Stream<List<WiFiAccessPoint>>? _onScannedResultsAvailable;
+  Stream<Result<List<WiFiAccessPoint>, GetScannedResultsErrors>>?
+      _onScannedResultsAvailable;
 
-  /// Checks if it is ok to invoke [startScan].
-  ///
-  /// Necesearry platform requirements, like permissions dependent services,
-  /// configuration, etc are checked.
+  /// Request a Wi-Fi scan.
   ///
-  /// Set [askPermissions] flag to ask user for necessary permissions.
-  Future<CanStartScan> canStartScan({bool askPermissions = true}) async {
-    final canCode = await _channel.invokeMethod<int>("canStartScan", {
+  /// Returns null if successful, else [StartScanErrors].
```

Comment:
I am more use to have the "OK" value part of the enum, but then C does not have null types, so...  
Both styles are fine to me, just not sure whether there exists some consensus on this.

Reply:
You mean having `StartScanErrors.none`, if no error instead of `null`?

Reply:
Yes. But this is fine as well.

Reply:
with that, it is hard to incorporate something like `hasError` check with generic `Result<ValueType, ResultType>` class.